### ISSUE TITLE
Changing to vi mode via config magic makes repl stop reacting to almost all keyboard input, but via cmdline arg works fine

### ISSUE THREAD

**Issue by chrisbarber** [ipython/ipython#12897]
Ubuntu 18.04.5 LTS
GNOME Terminal 3.28.2

```
$ python3 -m IPython
Python 3.6.9 (default, Jan 26 2021, 15:33:00) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %config TerminalInteractiveShell.editing_mode
Out[1]: 'emacs'

In [2]: %config TerminalInteractiveShell.editing_mode='vi'
```

Stops reacting to all keyboard input except Ctrl+d (at least that I can find)

```
python3 -m IPython --TerminalInteractiveShell.editing_mode=vi
```

works fine on the other hand.

**Comment by MrMino** [comment]
I can confirm this also happens on 7.22 / current master on CPython 3.8.5.

**Comment by MrMino** [comment]
Fixed in #13091.

**Comment by chrisbarber** [comment]
Pretty sure this is fixed / was meant to be auto-closed when #13121 was merged.
